doc/doxygen/src/board-cpu-features.md

```diff
@@ -0,0 +1,99 @@
+
+FEATURES                                                             {#features}
+========
+
+`FEATURES_PROVIDED` are hardware (including toolchain) features (e.g.:`periph_hwrng`,
+`periph_uart`) that are available or characteristics (e.g:`arch_32bits`) of a board.
+
+Providing a `FEATURE`
+--------------------
+
+For a `FEATURE` to be provided by a `board` it must meet 2 criteria, and for
+`periph` and other `hw` it must follow the 3rd criteria.
+
+- Needs the "hardware" or BSP support (toolchain)
+    - e.g.: `stm32l152re` has an SPI peripheral
+            `riotboot` needs to be able to link and flash at an offset
+- Needs support in RIOT, an implementation of an api to interact with the `hw`
+    - e.g.: `cpu/stm32_common/periph/spi.c` is implemented for `stm32l1`
+            `riotboot` needs an implementation of `cpu_jump_to_image`
+- Wiring between the `cpu/soc` a `bus` and other `cpu`/`hw` components.
```

Comment:
For this one `riotboot` could also be included as it shows a non directly hardware thing but needs BSP/build system "wiring" with the usage of an adapted linker script, RIOT specific flasher characteristics, slots configuration https://github.com/RIOT-OS/RIOT/blob/15de138674652b3d83fa75a01d01efa91052aa33/cpu/cortexm_common/Makefile.include#L26-L71

Reply:
Mentioned on the first point!